### PR TITLE
Mono: Remove remoting code.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
The first commit just removes DISABLE_REMOTING blocks, the second commit cleans up leftover code.

**Comment by naricc** [comment]
What is remoting code and why are we removing it? Is there an associated issue?

**Comment by marek-safar** [comment]
https://docs.microsoft.com/en-us/dotnet/core/porting/net-framework-tech-unavailable#remoting

File: src/mono/mono/metadata/object.h [hunk 1]
```diff
@@ -301,8 +301,6 @@ mono_runtime_exec_main	    (MonoMethod *method, MonoArray *args,
 MONO_API MONO_RT_EXTERNAL_ONLY int
 mono_runtime_set_main_args  (int argc, char* argv[]);
 
-/* The following functions won't be available with mono was configured with remoting disabled. */
-/*#ifndef DISABLE_REMOTING */
```
Comment (lambdageek) [inline src/mono/mono/metadata/object.h:305]: We should keep some variant of this comment: otherwise it's not obvious why there's an apparent API break

**Review by lambdageek — APPROVED** [review]

**Pull request merged**